confgenerator/logging_modify_fields.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confgenerator
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/filter"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+)
+
+type ModifyField struct {
+	// Source of value for this field
+	MoveFrom    string  `yaml:"move_from" validate:"omitempty,field,excluded_with=CopyFrom StaticValue"`
+	CopyFrom    string  `yaml:"copy_from" validate:"omitempty,field,excluded_with=MoveFrom StaticValue"`
+	StaticValue *string `yaml:"static_value" validate:"excluded_with=MoveFrom CopyFrom"`
```

Comment:
```suggestion
        // Use "*string" instead of "string" here so the default is nil instead of an empty config.
        // This is required to allow users to specify an empty string as the config value.
	StaticValue *string `yaml:"static_value" validate:"excluded_with=MoveFrom CopyFrom"`
```

Reply:
Hm, I don't want to put this here and interrupt the flow of the code for this note. Using `*string` in this way is a common convention - e.g. every protobuf does this. Can I get away with leaving out a comment for this?